CriticalMass/AsyncOperation.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  AsyncOperation.swift
+//  CriticalMaps
+//
+//  Created by Илья Глущук on 13/10/2019.
+//  Copyright © 2019 Pokus Labs. All rights reserved.
+//
+
+import Foundation
+
+class AsyncOperation: Operation {
+    override var isAsynchronous: Bool {
+        return true
+    }
+    
+    private var _isFinished: Bool = false
+    
+    override var isFinished: Bool {
+        set {
+            willChangeValue(forKey: "isFinished")
```

Comment:
To track state maybe an enum would be nice:
```swift
enum State: String {
    case ready, executing, finished

    fileprivate var keyPath: String {
      return "is\(rawValue.capitalized)"
    }
  }
```

Reply:
also then
```swift
var state = State.ready {
  willSet {
    willChangeValue(forKey: newValue.keyPath)
    willChangeValue(forKey: state.keyPath)
  }
  didSet {
    didChangeValue(forKey: oldValue.keyPath)
    didChangeValue(forKey: state.keyPath)
  }
}
```

Reply:
Looks nice, I changed code to your suggestion.